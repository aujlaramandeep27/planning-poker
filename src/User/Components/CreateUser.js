import loginImg from '../../resources/images/learning_small.jpg';
import React from 'react';
import {connect} from 'react-redux';
import {PostUser} from '../Actions/UserActions';

export function CreateUser({postUser}) {

    return (
        <div className="uk-container uk-flex uk-flex-middle uk-height-viewport login">
            <div className="uk-animation-slide-left-medium uk-text-center"><h2>Welcome to planning poker</h2></div>
            <div className="uk-animation-scale-up uk-width-1-2@s uk-width-3-4@m"><img src={loginImg} alt="login"/></div>
            <div>
                <form onSubmit={event => {
                    PostNewUser(event, postUser)
                }}>
                    <div className="uk-margin">
                        <div className="uk-animation-slide-top uk-inline uk-width-1-1 uk-margin-small-top">
                            <span className="uk-form-icon" data-uk-icon="icon: user"/>
                            <input name="userInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <input className="uk-animation-slide-bottom uk-button uk-button-primary uk-width-1-1 uk-margin-small-top" type="submit" value="LOGIN / CREATE"/>
                    </div>
                </form>
            </div>
        </div>
    );

    function PostNewUser(event, postUser) {
        event.preventDefault();
        postUser(event.target.userInputName.value.trim());
    }

}

const MapDispatchToProps = dispatch => ({
    postUser: (username) => dispatch(PostUser(username)),
});

export default connect(null, MapDispatchToProps)(CreateUser)
