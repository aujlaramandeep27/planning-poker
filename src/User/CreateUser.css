.login-background {
    background-color: #37214d;
}

h1, h2, h3 {
    font-family: 'Frank Ruhl Libre', serif;
    color: #ffffff;
    margin: 2px;
}

.uk-button-primary {
    background-color: #a1dbff;
}