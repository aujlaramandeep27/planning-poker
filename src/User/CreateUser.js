import React from 'react';

export default class CreateUser extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: ''
        }
    }
    
    handleChange = (e) => {
        this.setState({
            name: e.target.value
        })
    }

    onSubmit = (e) => {
        e.preventDefault();
        const body = {
         name: this.state.name
        }
        // API call here
        console.log(body);
        this.props.history.push('/meetings');      
     }

    render() {
        return (
            <div className="uk-flex uk-flex-center">
                <div className="uk-card uk-card-default uk-card-body uk-width-1-2@m">
                    <h1 className="uk-heading-small">Create New User</h1>                
                    <input placeholder="Name" className="uk-input uk-margin" 
                        value={this.state.name} 
                        onChange={e => this.handleChange(e)}></input>
                    <button className="uk-button uk-button-primary"
                        onClick={(e) => this.onSubmit(e)}>CREATE</button>
                </div>
            </div>
        );
    }    
}