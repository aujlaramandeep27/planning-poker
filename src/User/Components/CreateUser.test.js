import {CreateUser} from './CreateUser';
import React from 'react';

describe('Container: User', () => {
    it('should render CreateUser correctly', () => {
        const dispatches = {
            goToMeetings: jest.fn()
        };

        const state = {
            user: {
                name: 'Mock User'
            },
            error: null
        };

        expect(CreateUser({...dispatches, ...state})).toMatchSnapshot();
    });
});