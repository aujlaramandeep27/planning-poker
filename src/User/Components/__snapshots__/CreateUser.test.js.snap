// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: User should render CreateUser correctly 1`] = `
<div
  className="uk-container uk-flex uk-flex-middle uk-height-viewport login"
>
  <div
    className="uk-animation-slide-left-medium uk-text-center"
  >
    <h2>
      Welcome to planning poker
    </h2>
  </div>
  <div
    className="uk-animation-scale-up uk-width-1-2@s uk-width-3-4@m"
  >
    <img
      alt="login"
      src="learning_small.jpg"
    />
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="uk-margin"
      >
        <div
          className="uk-animation-slide-top uk-inline uk-width-1-1 uk-margin-small-top"
        >
          <span
            className="uk-form-icon"
            data-uk-icon="icon: user"
          />
          <input
            className="uk-input"
            name="userInputName"
            placeholder="Name"
            type="text"
          />
        </div>
        <input
          className="uk-animation-slide-bottom uk-button uk-button-primary uk-width-1-1 uk-margin-small-top"
          type="submit"
          value="LOGIN / CREATE"
        />
      </div>
    </form>
  </div>
</div>
`;
